src/github/pagination.ts

```diff
@@ -1,13 +1,47 @@
-import { AnyResponse } from '@octokit/rest'
-import { GitHubAPI } from './'
+import Octokit from '@octokit/rest'
 
-export function addPagination (octokit: GitHubAPI) {
-  octokit.paginate = paginate.bind(null, octokit)
+// tslint:disable-next-line
+const octokitGetNextPage = require('octokit-pagination-methods/lib/get-next-page')
+// tslint:disable-next-line
+const octokitHasNextPage = require('octokit-pagination-methods/lib/has-next-page')
+
+export function addPagination (octokit: Octokit) {
+  const octokitPaginate = octokit.paginate
+
+  octokit.paginate = Object.assign(
+    (...args: any[]) => paginate(octokit, octokitPaginate, args[0], args[1], args[2]),
+    { iterator: octokit.paginate.iterator }
+  )
 }
 
-const defaultCallback = (response: AnyResponse, done?: () => void) => response
+const defaultCallback = (response: Octokit.AnyResponse, done?: () => void) => response
+
+async function paginate (octokit: Octokit, octokitPaginate: Octokit.Paginate, ...args: any[]) {
+  // Until we fully deprecate the old paginate method, we need to check if the first argument is a promise
+  // If it's not, we return the old function signature
```

Comment:
that comment is wrong, we actually return the old signature only if the first argument _is_ a promise